tests/test_message.py

```diff
@@ -4,17 +4,49 @@
 
 # test url detection
 
-def test_url_detection_of_https():
+def test_basic_url_detection_of_https():
     expected = "https://example.org"
+    assert message.extract_url("foo <https://example.org>") == expected
 
-    assert message.extract_url("foo https://example.org") == expected
+def test_hash_in_url():
+    expected = "https://#.org"
```

Comment:
I understand that there is no need to enforce strictly correct URLs at the regex level but at least the tests should show valid URLs.

So, I'd rather use

```
https://example.org#foo
```